[BUG] Failed to cmake the knnlib on macOS

We are running this on macOS 12.4.

```
+ cmake .
-- The C compiler identification is AppleClang 13.1.6.13160021
-- The CXX compiler identification is AppleClang 13.1.6.13160021
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Deprecation Warning at external/nmslib/similarity_search/CMakeLists.txt:11 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- Build type: Release
-- Header files: /private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/cmd_options.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/distcomp.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/distcomp_edist.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/eval_metrics.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/eval_results.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/experimentconf.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/experiments.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/factory/init_methods.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/factory/init_spaces.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/falconn_heap_mod.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/flexbuff.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/floatdiff.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/global.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/gold_standard.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/idtype.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/incremental_quick_select.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/index.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/init.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/inmem_inv_index.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/knnquery.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/knnqueue.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/logging.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/memory.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/meta_analysis.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/method/dummy.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/method/hnsw.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/method/hnsw_distfunc_opt_impl_inline.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/method/permutation_prefix_index.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/method/pivot_neighb_common.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/method/pivot_neighb_invindx.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/method/seqsearch.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/method/simple_inverted_index.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/method/small_world_rand.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/method/vptree.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/method/vptree_utils.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/methodfactory.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/my_isnan_isinf.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/object.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/params.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/params_cmdline.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/params_def.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/permutation_type.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/permutation_utils.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/portable_align.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/portable_intrinsics.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/portable_popcount.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/portable_prefetch.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/portable_simd.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/ported_boost_progress.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/pow.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/query.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/query_creator.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/rangequery.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/read_data.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/report.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/report_intr_dim.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/searchoracle.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/simddebug.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/sort_arr_bi.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/space.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/space/space_ab_diverg.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/space/space_bit_hamming.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/space/space_bit_jaccard.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/space/space_bit_vector.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/space/space_bregman.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/space/space_dummy.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/space/space_js.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/space/space_l2sqr_sift.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/space/space_leven.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/space/space_lp.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/space/space_rank_correl.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/space/space_renyi_diverg.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/space/space_scalar.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/space/space_sparse_bin_common.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/space/space_sparse_dense_fusion.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/space/space_sparse_jaccard.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/space/space_sparse_lp.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/space/space_sparse_scalar.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/space/space_sparse_scalar_bin_fast.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/space/space_sparse_scalar_fast.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/space/space_sparse_vector.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/space/space_sparse_vector_inter.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/space/space_sqfd.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/space/space_string.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/space/space_vector.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/space/space_vector_gen.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/space/space_word_embed.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/spacefactory.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/thread_pool.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/tune.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/utils.h;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/include/ztimer.h
-- Source files: /private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/distcomp_bregman.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/distcomp_diverg.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/distcomp_edist.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/distcomp_js.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/distcomp_l2sqr_sift.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/distcomp_lp.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/distcomp_overlap.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/distcomp_rankcorr.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/distcomp_scalar.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/distcomp_sparse_scalar_fast.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/experimentconf.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/global.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/init.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/knnquery.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/logging.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/memory.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/method/dummy.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/method/hnsw.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/method/hnsw_distfunc_opt.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/method/pivot_neighb_invindx.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/method/seqsearch.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/method/simple_inverted_index.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/method/small_world_rand.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/method/vptree.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/params.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/params_cmdline.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/query.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/rangequery.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/searchoracle.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/space.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/space/space_ab_diverg.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/space/space_bregman.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/space/space_dummy.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/space/space_js.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/space/space_l2sqr_sift.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/space/space_lp.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/space/space_renyi_diverg.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/space/space_scalar.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/space/space_sparse_dense_fusion.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/space/space_sparse_jaccard.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/space/space_sparse_lp.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/space/space_sparse_scalar_bin_fast.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/space/space_sparse_scalar_fast.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/space/space_sparse_vector.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/space/space_sparse_vector_inter.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/space/space_string.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/space/space_vector.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/space/space_word_embed.cc;/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/external/nmslib/similarity_search/src/utils.cc
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Found Threads.
-- OS:                Darwin
-- Compiler:          Clang 13.1.6
-- CXX flags release: -Wall -Wunreachable-code -Wcast-align -O3 -DNDEBUG -std=c++11 -DHAVE_CXX0X -pthread -fpic  -march=x86-64
-- CXX flags debug:   -Wall -Wunreachable-code -Wcast-align -ggdb  -DNDEBUG -std=c++11 -DHAVE_CXX0X -pthread -fpic  -march=x86-64
-- Found OpenMP_C: -Xpreprocessor -fopenmp  
-- Found OpenMP_CXX: -Xpreprocessor -fopenmp -I/usr/local/opt/libomp/include  
-- Found OpenMP: TRUE   
-- Found ZLIB: /Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/lib/libz.tbd (found version "1.2.11") 
-- Looking for sgemm_
-- Looking for sgemm_ - not found
-- Looking for dgemm_
-- Looking for dgemm_ - found
-- Found BLAS: /Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/System/Library/Frameworks/Accelerate.framework  
-- The Fortran compiler identification is unknown
CMake Error at CMakeLists.txt:112 (enable_language):
  No CMAKE_Fortran_COMPILER could be found.

  Tell CMake where to find the compiler by setting either the environment
  variable "FC" or the CMake cache entry CMAKE_Fortran_COMPILER to the full
  path to the compiler, or to the compiler name if it is in the PATH.


-- Configuring incomplete, errors occurred!
See also "/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/CMakeFiles/CMakeOutput.log".
See also "/private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/jni/CMakeFiles/CMakeError.log".
2022-08-04 17:17:42 ERROR    Error building k-NN, retry with: ./build.sh manifests/2.2.0/opensearch-2.2.0.yml --component k-NN --snapshot
Traceback (most recent call last):
  File "./src/run_build.py", line 81, in <module>
    sys.exit(main())
  File "./src/run_build.py", line 68, in main
    builder.build(build_recorder)
  File "/private/var/jenkins/workspace/distribution-build-opensearch/src/build_workflow/builder_from_source.py", line 55, in build
    self.git_repo.execute(build_command)
  File "/private/var/jenkins/workspace/distribution-build-opensearch/src/git/git_repository.py", line 83, in execute
    subprocess.check_call(command, cwd=cwd, shell=True)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/subprocess.py", line 363, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command 'bash /private/var/folders/2g/jjj_bgln5qn8c6t1szz_fmmc0000gn/T/tmp7k6buiz4/k-NN/scripts/build.sh -v 2.2.0 -p darwin -a x64 -s true -o builds' returned non-zero exit status 1.

```

--- a/DEVELOPER_GUIDE.md
+++ b/DEVELOPER_GUIDE.md
@@ -3,6 +3,9 @@
     - [Fork OpenSearch k-NN Repo](#fork-opensearch-k-nn-repo)
     - [Install Prerequisites](#install-prerequisites)
       - [JDK 11](#jdk-11)
+      - [CMake](#cmake)
+      - [Faiss Dependencies](#Faiss-Dependencies)
+      - [Environment](#Environment)
   - [Use an Editor](#use-an-editor)
     - [IntelliJ IDEA](#intellij-idea)
   - [Build](#build)
@@ -55,10 +58,25 @@ In addition to this, the plugin has been tested with JDK 17, and this JDK versio
 
 The plugin requires that cmake >= 3.17.2 is installed in order to build the JNI libraries.
 
+One easy way to install on mac or linux is to use pip:
+```bash
+pip install cmake==3.17.2
+```
+
 #### Faiss Dependencies
 
 To build the *faiss* JNI library, you need to have openmp, lapack and blas installed. For more information on *faiss* 
 dependencies, please refer to [their documentation](https://github.com/facebookresearch/faiss/blob/main/INSTALL.md).
+
+[Openblas](https://www.openblas.net/) can be used for both lapack and blas. To install on Mac, run:
+```bash
+brew install openblas
+```
+
+Additionally, the `gcc` toolchain needs to be installed on Mac. To install, run:
+```bash
+brew install gcc
+```
 
 #### Environment
 

